From 6de3ce2ca66e0d253f0df51742d53d0367ae4b3f Mon Sep 17 00:00:00 2001
From: Roman Bulgakov <roman.bulgakov@globallogic.com>
Date: Fri, 7 Aug 2015 17:35:45 +0300
Subject: [PATCH 069/142] FogBugz #316846-4: docs: dts-bindings: add doc for
 Arria 10 NAND ECC

This patch adds the Altera NAND Flash Controller
ECC device tree documentation to describe Arria 10
NAND EDAC entries.

Signed-off-by: Roman Bulgakov <roman.bulgakov@globallogic.com>
Signed-off-by: Dinh Nguyen <dinguyen@opensource.altera.com>
[Original patch taken from
 https://github.com/altera-opensource/linux-socfpga.git
 tag rel_socfpga-4.8_17.03.01_pr]
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 .../bindings/arm/altera/socfpga-nand-ecc.txt       | 33 ++++++++++++++++++++++
 1 file changed, 33 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/arm/altera/socfpga-nand-ecc.txt

diff --git a/Documentation/devicetree/bindings/arm/altera/socfpga-nand-ecc.txt b/Documentation/devicetree/bindings/arm/altera/socfpga-nand-ecc.txt
new file mode 100644
index 000000000000..7031e611edde
--- /dev/null
+++ b/Documentation/devicetree/bindings/arm/altera/socfpga-nand-ecc.txt
@@ -0,0 +1,33 @@
+Altera SoCFPGA NAND Flash Controller FIFOs Error Detection and Correction [EDAC]
+
+The Altera Arria 10 NAND flash controller has three local SRAM memory buffers:
+the write FIFO buffer, the read FIFO buffer and the ECC buffer.
+Each of these memories is protected by error checking and correction (ECC),
+and by interrupts for single and double-bit errors.
+
+NAND ECC Required Properties:
+- compatible : Should be "altr,a10-nand-buf-edac" for the ECC buffer
+	or "altr,a10-nand-rd-edac" for the read FIFO buffer
+	or "altr,a10-nand-wr-edac" for the write FIFO buffer.
+- reg : Address and size for ECC error interrupt status and clear registers.
+- interrupts : Should be single bit error interrupt, then double bit error
+	interrupt.
+
+Example:
+	nandedac_buf@ff8c2000 {
+		compatible = "altr,a10-nand-buf-edac";
+		reg = <0xff8c2000 0x400>;
+		interrupts = <0 2 4>, <0 0 4>;
+	};
+
+	nandedac_rd@ff8c2400 {
+		compatible = "altr,a10-nand-rd-edac";
+		reg = <0xff8c2400 0x400>;
+		interrupts = <0 2 4>, <0 0 4>;
+	};
+
+	nandedac_wr@ff8c2800 {
+		compatible = "altr,a10-nand-wr-edac";
+		reg = <0xff8c2800 0x400>;
+		interrupts = <0 2 4>, <0 0 4>;
+	};
-- 
2.11.0

